addons/mail/models/mail_thread.py

```diff
@@ -2009,14 +2009,22 @@ def message_post(self, *,
                  )
             )
         attachment_ids = list(attachment_ids or [])
-        if partner_ids and not tools.is_list_of(partner_ids, int):
-            raise ValueError(
-                _('Posting a message should receive partners as a list of IDs (received %(pids)s)',
-                  pids=repr(partner_ids),
-                 )
-            )
         partner_ids = list(partner_ids or [])
 
+
+        # def makePartner(partner):
+        #     if isinstance(partner, str):
+        #         partner = self.env['res.partner'].create({ "name": partner, "email": partner })
+        #         return partner.id
+
+
+        # partner_ids = list(map(makePartner, partner_ids))
+
+        partner_emails = [email for email in partner_ids if isinstance(email, str)]
+        partner_ids = map(lambda x: x.id, partner_ids + self.env['res.partner'].create([
```

Comment:
you can't concatenate a list (`partner_ids`) to a recordset (the result of the create), and the lambda function would not work on `partner_ids`

instead, you can use the `ids` attribute of the recordset like this:

  ```py
  partner_ids + self.env['res.partner'].create([{"name": email, "email": email} for email in partner_emails]).ids
```

but wouldn't it be easier to just pass an ids array and an email array from the js?

Reply:
> but wouldn't it be easier to just pass an ids array and an email array from the js?

I m not sure, it will create more diff and duplicate some logic just for this. I think it's not that bad to have some string inside this list.